tests/main/lxd/task.yaml

```diff
@@ -141,17 +125,34 @@ execute: |
     lxd.lxc exec my-nesting-ubuntu -- lxd waitready
     lxd.lxc exec my-nesting-ubuntu -- lxd init --auto
 
-    (
-        . /etc/os-release
-        # TODO:2004: remove version check once the image for ubuntu 20.04 is published
-        if [ "$VERSION_ID" == 20.04 ]; then
-            # shellcheck disable=SC2031
-            VERSION_ID=19.10
-        fi
-        lxd.lxc exec my-nesting-ubuntu -- lxd.lxc launch "ubuntu:${VERSION_ID:-}" my-inner-ubuntu
-    )
+    lxd.lxc exec my-nesting-ubuntu -- lxd.lxc launch "ubuntu:${VERSION_ID:-}" my-inner-ubuntu
     lxd.lxc exec my-nesting-ubuntu -- lxd.lxc exec my-inner-ubuntu -- echo "from-the-INSIDE-inside" | MATCH from-the-INSIDE-inside
 
+    # finally check that we can't run snapd inside a nested lxd container as 
+    # current apparmor does not support this, so if this works it is probably a
+    # confinement bug
+    lxd.lxc exec my-nesting-ubuntu -- lxd.lxc file push --quiet prep-snapd-in-lxd.sh my-inner-ubuntu/root/
+    snapdDeb=$(lxd.lxc exec my-nesting-ubuntu -- sh -c 'ls snapd_*.deb')
+    lxd.lxc exec my-nesting-ubuntu -- lxd.lxc file push --quiet $snapdDeb my-inner-ubuntu/root/
+    echo "Setting up proxy for the *inside nested* container"
+    if [ -n "${http_proxy:-}" ]; then
+        lxd.lxc exec my-nesting-ubuntu -- \
+            lxd.lxc exec my-inner-ubuntu -- \
+                sh -c "echo http_proxy=$http_proxy >> /etc/environment"
+    fi
+    if [ -n "${https_proxy:-}" ]; then
+        lxd.lxc exec my-nesting-ubuntu -- \
+            lxd.lxc exec my-inner-ubuntu -- \
+                sh -c "echo https_proxy=$https_proxy >> /etc/environment"
+    fi
+    lxd.lxc exec my-nesting-ubuntu -- \
+        lxd.lxc exec my-inner-ubuntu -- \
+            /bin/bash /root/prep-snapd-in-lxd.sh
+
+    lxd.lxc exec my-nesting-ubuntu -- \
+        lxd.lxc exec my-inner-ubuntu -- \
+            snap install snap install test-snapd-sh 2>&1 | MATCH error: system does not fully support snapd: apparmor detected but insufficient permissions to use it
```

Comment:
I think you meant just `snap install test-snapd-sh ...`

Reply:
/o\ yes indeed haha